src/utils/yang_baxter.jl

```diff
@@ -0,0 +1,48 @@
+# TODO: handling singularity
+
+change_direction(α) = log(-tanh(α/2))
+
+"""
+    yang_baxter(α1, β1, γ1)
+
+Return α2, β2, γ2 according to the Yang-Baxter equation:
+
+    \\    /    |       |    \\    / 
+     \\  /     |       |     \\  /
+      α1      |       |      α2
+     |  \\    /         \\    /  |
+     |   \\  /           \\  /   |
+     |    β1      =      β2    |
+     |   /  \\           /  \\   |
+     |  /    \\         /    \\  |
+      γ1      |       |      γ1 
+     /  \\     |       |     /  \\
+    /    \\    |       |    /    \\
+"""
+function yang_baxter(α1, β1, γ1)
```

Comment:
can we rename this to `yan_baxter_param` or sth? I think we will later need this name for our rewrite rule of this.